packages/twenty-front/src/modules/action-menu/components/ActionMenuEffect.tsx

```diff
@@ -0,0 +1,41 @@
+import { useActionMenu } from '@/action-menu/hooks/useActionMenu';
+import { contextStoreTargetedRecordIdsState } from '@/context-store/states/contextStoreTargetedRecordIdsState';
+import { isDropdownOpenComponentState } from '@/ui/layout/dropdown/states/isDropdownOpenComponentState';
+import { extractComponentState } from '@/ui/utilities/state/component-state/utils/extractComponentState';
+import { useEffect } from 'react';
+import { useRecoilValue } from 'recoil';
+
+type ActionMenuEffectProps = {
+  actionMenuId: string;
+};
+
+export const ActionMenuEffect = ({ actionMenuId }: ActionMenuEffectProps) => {
+  const contextStoreTargetedRecordIds = useRecoilValue(
+    contextStoreTargetedRecordIdsState,
+  );
+
+  const { openActionBar, closeActionBar } = useActionMenu(actionMenuId);
+
+  const isDropdownOpen = useRecoilValue(
+    extractComponentState(
+      isDropdownOpenComponentState,
+      `action-menu-dropdown-${actionMenuId}`,
+    ),
+  );
+
+  useEffect(() => {
+    if (contextStoreTargetedRecordIds.length > 0 && !isDropdownOpen) {
+      openActionBar();
```

Comment:
add Comment TODO here

Reply:
explain why we take care the ActionMenuBar but not the Dropdown (because the dropdown is already handled by sync handlers as we have the events (right click to open, click outside to close)